opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -1799,48 +1805,35 @@ export const elFindByIds = async (context, user, ids, opts = {}) => {
         script_field_denormalization_count: REL_COUNT_SCRIPT_FIELD
       };
     }
-    let searchAfter;
-    let hasNextPage = true;
-    while (hasNextPage) {
-      if (searchAfter) {
-        body.search_after = searchAfter;
-      }
-      const _source = { excludes: [] };
-      if (withoutRels) _source.excludes.push(`${REL_INDEX_PREFIX}*`);
-      if (baseData) _source.includes = [...BASE_FIELDS, ...baseFields];
-      const query = {
-        index: computedIndices,
-        size: ES_MAX_PAGINATION,
-        track_total_hits: false,
-        _source,
-        body,
-      };
-      if (withoutRels) { // Force denorm rel security
-        query.docvalue_fields = REL_DEFAULT_FETCH;
-      }
-      logApp.debug('[SEARCH] elInternalLoadById', { query });
-      const searchType = `${ids} (${types ? types.join(', ') : 'Any'})`;
-      const data = await elRawSearch(context, user, searchType, query).catch((err) => {
-        throw DatabaseError('Find direct ids fail', { cause: err, query, searchType });
-      });
-      const elements = data.hits.hits;
-      if (elements.length > workingIds.length) {
-        logApp.warn('Search query returned more elements than expected', { ids: workingIds });
-      }
-      if (elements.length > 0) {
-        const convertedHits = await elConvertHits(elements);
-        hits.push(...convertedHits);
-        if (elements.length < ES_MAX_PAGINATION) {
-          hasNextPage = false;
-        } else {
-          const { sort } = elements[elements.length - 1];
-          searchAfter = sort;
-          hasNextPage = true;
-        }
-      } else {
-        hasNextPage = false;
+    const _source = { excludes: [] };
+    if (withoutRels) _source.excludes.push(`${REL_INDEX_PREFIX}*`);
+    if (baseData) _source.includes = [...BASE_FIELDS, ...baseFields];
+    const query = {
+      index: computedIndices,
+      size: ES_MAX_PAGINATION,
+      track_total_hits: false,
+      _source,
+      body,
+    };
+    if (withoutRels) { // Force denorm rel security
+      query.docvalue_fields = REL_DEFAULT_FETCH;
+    }
+    logApp.debug('[SEARCH] elInternalLoadById', { query });
+    const searchType = `${ids} (${types ? types.join(', ') : 'Any'})`;
+    const data = await elRawSearch(context, user, searchType, query).catch((err) => {
+      throw DatabaseError('Find direct ids fail', { cause: err, query, searchType });
+    });
+    const elements = data.hits.hits;
+    if (elements.length > workingIds.length) {
+      logApp.info('Search query returned more elements than expected', { ids: workingIds });
+      if (elements.length >= ES_MAX_PAGINATION) {
+        throw DatabaseError('Ids loading returned more elements than paging allowed for, some elements could not be loaded', { ids: workingIds });
       }
```

Comment:
we could add  workingIds.length elements.length and ES_MAX_PAGINATION  in log and exceptions details.
i wouldn't add the full list of workingIds, maybe we could return only the list of ids that have some duplicates ?